common/v2/features/Dashboard/Settings/Settings.tsx

```diff
@@ -1,57 +1,130 @@
+            addressBook={addressBook}
+            toggleFlipped={toggleFlipped}
+            deleteAddressBooks={deleteAddressBooks}
+          />
+        )
+      }
+    </FlippablePanel>
+  );
+}
+
+function renderGeneralSettingsPanel() {
+  const { updateSettings, settings } = useContext(SettingsContext);
+  return <GeneralSettings updateGlobalSettings={updateSettings} globalSettings={settings} />;
+}
+
+interface TabOptions {
+  [key: string]: React.ReactNode;
+}
+
+function renderMobile() {
+  const [tab, setTab] = useState('wallets');
+  const tabOptions: TabOptions = {
+    ['wallets']: renderAccountPanel(),
+    ['addresses']: renderAddressPanel(),
+    ['general']: renderGeneralSettingsPanel()
+  };
+  const currentTab = tabOptions[tab];
+  return (
+    <>
+      <Tabs>
+        <a href="#" onClick={() => setTab('wallets')}>
+          Your Wallets
+        </a>
+        <a href="#" onClick={() => setTab('addresses')}>
+          Addresses
+        </a>
+        <a href="#" onClick={() => setTab('general')}>
+          General
+        </a>
+      </Tabs>
+      <SettingsContent>
+        <SettingsHeading>
+          <SettingsHeadingIcon src={settingsIcon} alt="Settings" />
+          {translate('SETTINGS_HEADING')}
+        </SettingsHeading>
+        {currentTab}
+      </SettingsContent>
+    </>
+  );
+}
+
+function renderDesktop() {
   return (
-    <Layout className="Settings">
-      <Heading className="Settings-heading">
-        <img src={settingsIcon} alt="Settings" className="Settings-heading-icon" />
-        Settings
-      </Heading>
-      <AccountContext.Consumer>
-        {({ accounts, deleteAccount }) => (
-          <FlippablePanel>
-            {({ flipped }) =>
-              flipped ? (
-                <p>Add Account</p>
-              ) : (
-                <AccountList accounts={accounts} deleteAccount={deleteAccount} />
-              )
-            }
-          </FlippablePanel>
-        )}
-      </AccountContext.Consumer>
-      <AddressBookContext.Consumer>
-        {({ createAddressBooks, addressBook, deleteAddressBooks }) => (
-          <FlippablePanel>
-            {({ flipped, toggleFlipped }) =>
-              flipped ? (
-                <AddToAddressBook
-                  toggleFlipped={toggleFlipped}
-                  createAddressBooks={createAddressBooks}
-                />
-              ) : (
-                <AddressBook
-                  addressBook={addressBook}
-                  toggleFlipped={toggleFlipped}
-                  deleteAddressBooks={deleteAddressBooks}
-                />
-              )
-            }
-          </FlippablePanel>
-        )}
-      </AddressBookContext.Consumer>
-      <GeneralSettings />
-    </Layout>
+    <SettingsContent>
+      <SettingsHeading>
+        <SettingsHeadingIcon src={settingsIcon} alt="Settings" />
+        {translate('SETTINGS_HEADING')}
+      </SettingsHeading>
+      {renderAccountPanel()}
+      {renderAccountPanel()}
```

Comment:
It's rendering the AccountPanel twice! (most probably my mistake)
Should be renderAddressPanel